StripePaymentSheet/StripePaymentSheet/PaymentSheet/Intent.swift

```diff
@@ -199,18 +201,28 @@ extension STPConfirmPaymentMethodOptions {
         paymentMethodType: STPPaymentMethodType,
         customer: PaymentSheet.CustomerConfiguration?
     ) {
-        // This property cannot be set if there is no customer.
+        // Something went wrong if we're trying to save and there's no Customer!
         assert(!(shouldSave && customer == nil))
 
-        // Only support card and US bank setup_future_usage in payment_method_options
-        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount
-        else {
+        // This property should only be set if
+        // 1. We're displaying a "Save this pm for future payments" checkbox
+        // 2. The PM type is card or US bank
+        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount else {
+            return
+        }
+        // Passing "none" *overrides* the top-level setup_future_usage and is not what we want, b/c this code is called even when we don't display the "save" checkbox (e.g. when the PI top-level setup_future_usage is already set).
+        // Instead, we pass an empty string to 'unset' this value. This makes the PaymentIntent *inherit* the top-level setup_future_usage.
+        let sfuValue = shouldSave ? "off_session" : ""
+        switch paymentMethodType {
+        case .card:
+            cardOptions = cardOptions ?? STPConfirmCardOptions()
+            cardOptions?.additionalAPIParameters["setup_future_usage"] = sfuValue
+        case .USBankAccount:
+            usBankAccountOptions = usBankAccountOptions ?? STPConfirmUSBankAccountOptions(setupFutureUsage: .none)
+            usBankAccountOptions?.additionalAPIParameters["setup_future_usage"] = sfuValue
```

Comment:
So now we are setting the sfu param under the card/us bank account options? does this have the same effect?

Reply:
The request parameters end up the same as before.

The /confirm parameters dictionary looks like this:
```
▿ 3 elements
  ▿ 0 : 2 elements
    - key : "payment_method_options"
    ▿ value : 1 element
      ▿ 0 : 2 elements
        - key : card
        ▿ value : 2 elements
          ▿ 0 : 2 elements
            ▿ key : setup_future_usage
            - value : 
          ▿ 1 : 2 elements
            - key : moto
            - value : 1
  ▿ 1 : 2 elements
    - key : "client_secret"
    - value : pi_1Cl15wIl4IdHmuTbCWrpJXN6_secret_EkKtQ7Sg75hLDFKqFG8DtWcaK
  ▿ 2 : 2 elements
    - key : "payment_method"
    - value : pm_123456789

```